workload_generator/workload_generator.py

```diff
@@ -0,0 +1,30 @@
+import threading
+import parser
+import worker
+
+class WorkloadGenerator:
+
+    def __init__ (self, user_cmd_file):
+        self.user_cmd_file = user_cmd_file
+    
+    def update_file(user_cmd_file):
+        self.user_cmd_file = user_cmd_file
+
+    def run(self):
+        user_cmds = parser.parse_commands_file(self.user_cmd_file)
+        all_the_threads = {}
+        for user_id in user_cmds:
```

Comment:
Since we only have to support a single user for the first milestone this is good, but as we start to support a larger number of users, spawning a new thread for each might be too resource intensive. Maybe having a fixed set of worker threads and submitting tasks via the ThreadpoolExecutor (https://tutorialedge.net/python/concurrency/python-threadpoolexecutor-tutorial/) is an option. Just something to look into.

Reply:
Yea this is the problem with python, is that it probably maxes out at less than 1000 users. Golang on the other hand can handle around a million goroutines at the same time.

Reply:
So the one thing I wanna point out here is, we know the exact number of users there will be because we pre-process the user commands file before we spawn the worker threads. So I'm not sure the difference between starting that number of threads in a threadpool vs separately.

Unless you are suggesting we can use less threads than users to accomplish the task. Because at time 0, when the commands file has been processed all users are ready to start sending commands.

Reply:
I've looked into the thread pool executor already, since we last talked Isaac. I just want to be a little convinced of its value